smack-core/src/main/java/org/jivesoftware/smack/SmackFuture.java

```diff
@@ -162,7 +162,18 @@ public V getIfAvailable() {
         return result;
     }
 
-    protected final synchronized void maybeInvokeCallbacks() {
+    private boolean callbacksInvoked;
+
+    protected final synchronized void maybeInvokeCallbacksOnEvent() {
```

Comment:
I'm having trouble setting up my environment to build Smack, so this is only based on my reading of the code.

I think a `.onSuccess(...).onError(...)` call after the future has completed will still result in the success callback being called twice.

Reply:
Fair point. I've just pushed a revision that should fix this.